src/ui/handler/drag_pan.js

```diff
@@ -190,6 +190,10 @@ class DragPanHandler {
     _ignoreEvent(e) {
         const map = this._map;
 
+        // if boxZoom is disabled and then re-enabled, the drag_pan event listener for mousedown will fire before
+        // the boxZoom event listener, so map.boxZoom.isActive() will return false resulting in an erroneous
+        // drag on mousemove once the boxZoom is released, so we need this additional check.
+        if (map.boxZoom && e.shiftKey && e.button === 0) return true;
```

Comment:
Oh dear 🤦‍♂️.  I swear I really did read that comment , but then got turned upside-down thinking about event logic...

What made me ask was thinking about something like:
 - mousdown: start dragpan
 - press (and hold) shift
 - mouseup: i think the expectation would be to stop panning, but this check would  prevent that from happening (if I'm reading it right?)